documentation/modules/cruise-control/con-optimization-proposals.adoc

```diff
@@ -0,0 +1,82 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='con-optimization-proposals-{context}']
+
+= Optimization proposals overview
+
+An _optimization proposal_ is a summary of proposed changes that would produce a more balanced Kafka cluster, with partition workloads distributed more evenly among the brokers. 
+Each optimization proposal is based on the set of xref:con-optimization-goals-{context}[optimization goals] that were used to generate it.
+
+Optimization proposals always satisfy all the configured hard goals, although not necessarily all the soft goals. Hard goals are ignored if `skip_hard_goal_check: true` is specified in the `KafkaRebalance` custom resource.
+
+An optimization proposal contains the following information, in JSON, about the pending cluster rebalance:
+
+* An optimal mapping of partitions to brokers, generated by the Analyzer component of Cruise Control
+
+* Changes to the partition leaders
+
+* Statistics on the cluster rebalance (before and after)
+
+You can use the information to decide whether to implement the cluster rebalance, or change the optimization goals. 
+
+NOTE: All optimization proposals are dry runs: you cannot implement a cluster rebalance without first generating an optimization proposal. 
+There is no limit to the number of optimization proposals that can be generated.
+
+[discrete]
+== Contents of optimization proposals
+
+The following table explains the properties contained in an optimization proposal:
+
+[cols="2*",options="header",stripes="none",separator=¦]
+|===
+
+m¦JSON property
+¦Description
+
+m¦numIntraBrokerReplicaMovements
+¦Description
+
+m¦excludedBrokersForLeadership
+¦Description
+
+m¦numReplicaMovements
+¦Description
+
+m¦onDemandBalancednessScoreAfter
+¦Description
+
+m¦onDemandBalancednessScoreBefore
+¦Description
+
+m¦intraBrokerDataToMoveMB
+¦Description
+
+m¦recentWindows
+¦Description
+
+m¦dataToMoveMB
+¦Description
+
+m¦monitoredPartitionsPercentage
+¦Description
+
+m¦excludedTopics
+¦Description
+
+m¦numLeaderMovements
+¦Description
+
+m¦excludedBrokersForReplicaMove
+¦Description
```

Comment:
Some CC rebalance options allow for brokers to be excluded from the optimization proposals. These options are not currently supported so this list will be empty.